### ISSUE TITLE
Predicting Best Matching for new data

### ISSUE THREAD

**Issue by theSage21** [peterwittek/somoclu#39]
Hi,

Noob question here. Once I've got my map, how do I predict the best matching units for new data?

**Comment by xgdgsc** [comment]
http://chem-eng.utoronto.ca/~datamining/Presentations/SOM.pdf

**Comment by theSage21** [comment]
@xgdgsc Let me rephrase.  
How do I use somoclu to classify new data given an already existing  set of `.wts` and `.ux`?
Essentially what is the equivalent of `predict` given that `fit` is done via something like 
`somoclu -x 10 -y 10 input_file output_file`?

**Comment by Dringite** [comment]
In python prediction is relatively straight forward:

`W = som.codebook.reshape((som.codebook.shape[0] * som.codebook.shape[1], som.codebook.shape[2]))
X = data

D = -2*np.dot(W, X.T) + (W**2).sum(1)[:, None] + (X**2).sum(1)[:, None].T
BMU = (D==D.min(0)[None,:]).astype("float32").T
NBMU =  BMU.reshape((X.shape[0], som.codebook.shape[0], som.codebook.shape[1]))
new_bmus = np.vstack(NBMU.nonzero()[1:][::-1]).T`

X can be a single instance or as many as your RAM would allow.

**Comment by theSage21** [comment]
@Dringite thank you.
